### PR TITLE
add support to set aliase

### PR DESCRIPTION
**Pull request by mmailand**
add support to set aliase

### CODE REVIEW

**Comment by jerearista** [comment]
Jenkins, test this please

**Comment by jerearista** [comment]
Jenkins, retest this please

File: lib/rbeapi/api/alias.rb [hunk 1]
```diff
@@ -0,0 +1,151 @@
+#
+## Copyright (c) 2014, Arista Networks, Inc.
```
Comment (jerearista) [inline lib/rbeapi/api/alias.rb]: Please update the year to 2016

File: spec/system/rbeapi/api/alias_spec.rb [hunk 2]
```diff
@@ -0,0 +1,138 @@
+#
+# Copyright (c) 2015, Arista Networks, Inc.
```
Comment (jerearista) [inline spec/system/rbeapi/api/alias_spec.rb]: Update year

**Comment by mmailand** [comment]
changed as requested

File: spec/unit/rbeapi/api/alias/fixture_alias.text [hunk 3]
```diff
@@ -0,0 +1,3 @@
+alias Alias1 my command
```
Comment (devrobo) [inline spec/unit/rbeapi/api/alias/fixture_alias.text]: Does this addition handle multi-line aliases?

Comment (jerearista) [inline spec/unit/rbeapi/api/alias/fixture_alias.text]: Martin,
  Please ensure the regexes can handle multiline aliases as in the following:
```
!
prompt %H%R%v%P
alias sib show ip interface brief
no service configuration session max completed
no service configuration session max pending
!
alias desc
   !! Syntax: desc <interface> <description>
   1 conf
   2 int %1
   3 description %2
   4 end
!
```

One possible solution, while not terribly pretty, would be the following:
```
node.running_config.scan(/^alias (\w+)((?:(?= )(?:.+?)(?=\n)|\n(?:.+?)(?=\n\!)))/m)
=> [["sib", " show ip interface brief"],
 ["desc",
  "\n   !! Syntax: desc <interface> <description>\n   1 conf\n   2 int %1\n   3 description %2\n   4 end"]]
```

The initial \n in the multiline body is both a good indicator that it is a multiline alias.  You may still need to do a split before passing it to `node.config()` though.

Comment (mmailand) [inline spec/unit/rbeapi/api/alias/fixture_alias.text]: To be honest I am a bit stucked here.

I changed the regex and I can get a multiline alias (except for this extra space).

`
irb(main):004:0> al.getall
=> {"sis"=>{:name=>"sis", :command=>" show interface status"}, "desc"=>{:name=>"desc", :command=>"\n   !! Syntax: desc <interface> <description>\n   1 conf\n   2 int %1\n   3 description %2\n   4 end"}}
`

I can create a single line alias.

`
irb(main):005:0> al.create('sid', :command=>'show interface description')
`

But if I try a multiline one:

`
al.create('desc2', :command=>'\n   !! Syntax: desc <interface> <description>\n   1 conf\n   2 int %1\n   3 description %2\n   4 end')
`

The result in the FastCli running-conf looks like this.

`
alias desc2 \n !! Syntax: desc <interface> <description>\n 1 conf\n 2 int %1\n 3 description %2\n 4 end
`

Do you have any hints for me?

Comment (jerearista) [inline spec/unit/rbeapi/api/alias/fixture_alias.text]: I would suggest  in your `create` method, that you do something like 
```
cmd = ["alias #{name}"]
if command =~ /\n/
  command.split("\n").each { |a| cmd << a }
else  
  cmd[0] << command
end
configure(cmds)
```

File: lib/rbeapi/api/alias.rb [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+      #
+      # @example
+      #   {
+      #     alias: array<strings>
+      #   }
+      #
+      # @return [Hash<Symbol, Object>] Returns the alias resource as a hash
+      #   object from the nodes current configuration.
+      def get(name)
+        aliases = config.scan(/^alias #{name}((?:(?= )(?:.+?)(?=\n)|(?:\n(?:.+?))(?=\n\!)))/m)
```
Comment (mmailand) [inline lib/rbeapi/api/alias.rb]: this regex is still not perfect, I have an extra space in the beginning of the first capture group

File: lib/rbeapi/api/alias.rb [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+      #
+      # @param alias [Array] An array of values returned from the regular
+      #   expression scan of the aliass configuration.
+      #
+      # @return [Hash<Symbol, Object>] Returns the resource hash attribute.
+      def parse_alias_entry(name,command)
+        hsh = {}
+        hsh[:name] = name
+        com = command[0]
+        hsh[:command] = com.strip
```
Comment (mmailand) [inline lib/rbeapi/api/alias.rb]: Here I strip this extra space

File: lib/rbeapi/api/alias.rb (hunk 4 above)
Comment (mmailand) [inline lib/rbeapi/api/alias.rb]: this regex is still not perfect, I have an extra space in the beginning of the first capture group

**Comment by jerearista** [comment]
Jenkins, test this please

**Pull request merged**